include/kernel/KernelBuilder.h

```diff
@@ -88,6 +89,32 @@ namespace nalu{
     }
   }
 
+  template <template <typename> class T, typename... Args>
+  Kernel* build_face_elem_topo_kernel(int dimension,
+                                      stk::topology faceTopo, stk::topology elemTopo,
+                                      Args&&... args)
+  {
+    switch(faceTopo.value()) {
+      case stk::topology::QUAD_4:
+        return new T<AlgTraitsQuad4Hex8>(std::forward<Args>(args)...);
```

Comment:
We will need to think about this for the general face/elem approach as a quad4 can be attached to a wedge or pyramid. @rcknaus might have views here.